### ISSUE TITLE
netcdf with opendap: unresolved symbol (curl_easy_getinfo)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sjaaky74/netcdf4-python#70]
```
What steps will reproduce the problem?
1. compile netcdf 4.1.1 with opendap (default)
2. compile netCDF-0.9.3
3. in a python script, 'import netCDF4' fails because of an unresolved symbol 
(curl_easy_getinfo)

What is the expected output? What do you see instead?
a sucessful import, instead I got this error message

What version of the product are you using? On what operating system?
netCDF-0.9.3, Linux, Ubuntu 10.04, 64-bit

Please provide any additional information below.

For some reason, curl is not linked when creating netCDF4.so:

~/Download/netCDF4-0.9.3$ python setup.py
HDF5 found in /usr
netCDF4 found in /usr/local
[...]
gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions 
build/temp.linux-x86_64-2.6/netCDF4.o -L/usr/local/lib -Wl,-R/usr/local/lib 
-lnetcdf -lhdf5_hl -lhdf5 -lz -o build/lib.linux-x86_64-2.6/netCDF4.so
[...]

However, the curl dependency is well reported by nc-config
~/Download/netCDF4-0.9.3$ nc-config --libs
-L/usr/local/lib -lnetcdf -lhdf5_hl -lhdf5 -lz -lm -lcurl

Any help would be greatly apreciated. Thanks !

```

Original issue reported on code.google.com by `barth.al...@gmail.com` on 25 Oct 2010 at 8:08

**Comment by GoogleCodeExporter** [comment]
```
If you didn't build netcdf as a shared library, you need to explicitly link 
libcurl.  This is most easily done by cutting and pasting the failed link 
command into the terminal and adding -L<path to curl lib> -lcurl to the end.  
After that, run 'setup.py install' as normal.
```

Original comment by `whitaker.jeffrey@gmail.com` on 25 Oct 2010 at 8:19

**Comment by GoogleCodeExporter** [comment]
```
Thanks Jeffrey for your help. I needed to recompile NetCDF + HDF5 because on 
Ubuntu 10.04, it only included HDF5 1.8.4 (instead of 1.8.4-patch1) and I 
created a shared library. My problems with libcurl went thus away.

I'm wondering if it is not safer to rely on nc-config for the dependencies. For 
example in setyp.py:

dep = 
subprocess.Popen(['nc-config','--libs'],stdout=subprocess.PIPE).communicate()[0]
libs = [l[2:] for l in dep.split() if l[0:2] == '-l' ]
libdir = [l[2:] for l in dep.split() if l[0:2] == '-L' ]

But anyway, it worked now. Thanks again!

```

Original comment by `barth.al...@gmail.com` on 26 Oct 2010 at 9:17

**Comment by GoogleCodeExporter** [comment]
Original comment by `whitaker.jeffrey@gmail.com` on 26 Feb 2014 at 2:04
- Changed state: **Fixed**
